frontends/precinct-scanner/src/components/set_mark_thresholds_modal.test.tsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+
+import {
+  render,
+  fireEvent,
+  getByText as domGetByText,
+  waitFor,
+} from '@testing-library/react';
+import { electionSample } from '@votingworks/fixtures';
+
+import { SetMarkThresholdsModal } from './set_mark_thresholds_modal';
+
+test('renders warning message before allowing overrides to be set', () => {
+  const closeFn = jest.fn();
+  const { getByText } = render(
+    <SetMarkThresholdsModal
+      onClose={closeFn}
+      markThresholds={electionSample.markThresholds}
+      markThresholdOverrides={undefined}
+      setMarkThresholdOverrides={jest.fn()}
+    />
+  );
+  getByText('Override Mark Thresholds');
+  getByText(/WARNING: Do not proceed/);
+  fireEvent.click(getByText('Close'));
```

Comment:
I know this is copied over from frontends/bsd so not a big deal! We could use `screen` queries instead of the ones returned by `render` and `userEvent` instead of `fireEvent` to follow the latest practices:

```ts
screen.getByText('Override Mark Thresholds');
screen.getByText(/WARNING: Do not proceed/);
userEvent.click(screen.getByText('Close'));
```